### PR TITLE
[CPU] Fix ScaledDotProductAttension accuracy problem

### PR DESCRIPTION
**Pull request by luo-cheng2021**
### Details:
 - *Fix ScaledDotProductAttension accuracy problem*
   - Error: many models from timm repo failed to pass its accuracy test
   - Reason:
     - original attn_mask assumed that its shape was like [B, 1, L0, L0+L1], this was not true in encoder-based models, these shapes are possible: [B/1, H/1, L0/1, L0+L1], [H/1, L0/1, L0+L1] or scalar
     - k,v head number of model 'Rocketknight1/tiny-random-falcon-7b' is 1 which previously requires their head number must be equal to q

### Tickets:
 - *125517*

### CODE REVIEW

File: tests/model_hub_tests/torch_tests/test_timm.py [hunk 1]
```diff
@@ -75,7 +75,31 @@ def teardown_method(self):
 
     @pytest.mark.parametrize("name", ["mobilevitv2_050.cvnets_in1k",
                                       "poolformerv2_s12.sail_in1k",
-                                      "vit_base_patch8_224.augreg_in21k"])
+                                      "vit_base_patch8_224.augreg_in21k",
+                                      # TODO: will revert
```
Comment (mvafin) [inline tests/model_hub_tests/torch_tests/test_timm.py:79]: Please leave one of these models in the list after revert, so we will have it in precommit for future

**Review by mvafin — APPROVED** [review]

File: tests/model_hub_tests/torch_tests/test_timm.py (hunk 1 above)
Comment (luo-cheng2021) [reply tests/model_hub_tests/torch_tests/test_timm.py]: Done.

**Review by usstq — APPROVED** [review]
torch [scaled_dot_product_attention](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) supports arbitrary number of heads dimensions in q/k/v inputs : (N, ..., L, E) and thus the attention weight also become arbitrary dimensional (N, ..., L, S), and attn_mask must be broadcast-able to this shape. 
By using @ operator for Q*K' & W*V allows broadcasting on these head dimensions, this is flexible in practice for supporting various kv head sharing scheme - [multi-query or grouped-query](https://arxiv.org/pdf/2305.13245.pdf).

image

In multi-query case ,1 key/value head is shared among all heads of queries, so k/v cache must be indexed in broadcast way when being used in per-head kernel.


In grouped-query case, 5d q/k/v is required to feed into scaled_dot_product_attention:
 - q : (N, num_groups, q_heads_per_group, L, E)
 - k/v: (N, num_groups, 1, L, E)

one more related information on this, in some LLM like falcon/[chatGLM2](https://huggingface.co/THUDM/chatglm2-6b) using grouped-query, to simplify the logic, they didn't take advantage of SDPA's intrinsic support and manually broadcasted the k/v heads before feeding to attention layer, they only take advantage of grouped query to reduce kv-cache footprint, but ignored the memory access cost optimization opportunity.

LGTM, this PR adds support for multi-query, but grouped-query requires more work.

**Review by dmitry-gorokhov — APPROVED** [review]
